search/terms_aggregation.go

```diff
@@ -0,0 +1,201 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// TermsAggregation dynamically creates a bucket for each unique term of a field.
+// An empty TermsAggregation will have two issues with execution:
+//
+//   - the target Field must be non-null and non-empty.
+//   - a Size of 0 will return no buckets
+//
+// For more details see https://opensearch.org/docs/latest/opensearch/bucket-agg/
+type TermsAggregation struct {
+	// Field to be bucketed
+	Field string
+
+	// Size of the number of buckets to be returned. Negative sizes will be omitted
+	Size int
+
+	// Order list of [Order]s to sort the aggregation buckets. Default order is _count: desc
+	Order []Order
+
+	// Aggregations sub aggregations for each bucket. Mapped by string label to sub aggregation
+	Aggregations map[string]Aggregation
```

Comment:
can we change name Aggregations  to SubAggregations? it would self sufficient whats its means

Reply:
Calling it `Aggregations` is more inline with the OpenSearch request body which has the key as `aggs` rather than `subAggs`.

Reply:
I agree.  `Aggregations` is perfectly clear IMO